libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java

```diff
@@ -66,6 +67,25 @@ public void create(T item) {
         table.putItem(item);
     }
 
+    public void create(List<T> items) throws DataStoreException {
+        if (items.isEmpty()) {
+            LOGGER.info(LogHelper.buildLogMessage("No items found for creation"));
+            return;
+        }
+        var writeBatchBuilder = WriteBatch.builder(typeParameterClass).mappedTableResource(table);
+        items.forEach(writeBatchBuilder::addPutItem);
+        var batchWriteResult =
+                client.batchWriteItem(b -> b.writeBatches(writeBatchBuilder.build()));
+
+        if (!batchWriteResult.unprocessedPutItemsForTable(table).isEmpty()) {
+            throw new DataStoreException(
+                    String.format(
+                            "Unable to create %d datastore items",
+                            batchWriteResult.unprocessedPutItemsForTable(table).size()));
```

Comment:
Do we get any more information than this (i.e. _why_ it failed)? It looks like we're losing any context from the output except for the number of unprocessed items.

Reply:
(Possibly same question for the deletes - although at least that includes the IDs for the items)

Reply:
We don't - as it's generic.
One option would be to add a method to the `DynamodbItem` interface to enforce a useful way to get a string representation for any item that might be seen here?

Reply:
Hmm I see. It feels a little scary not to know what has/hasn't been created but it's also not totally clear what would be useful.

The DynamoDB documentation suggests you should just retry the request (with exponential backoff) - suggesting you shouldn't get any interesting errors. For 'normal' requests I think the AWS SDK does this automatically for us, but I'm not sure how it'll work with the bulk operation.

Reply:
hmm. might be a bit safer to just use the per item methods then, rather than having to worry about retrying things...